src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs

```diff
@@ -152,31 +173,64 @@ public async Task<IActionResult> ConfirmStandard(StandardVersionViewModel model,
+                await _applicationApiClient.UpdateStandardData(id, model.SelectedStandard.LarsCode, model.SelectedStandard.IFateReferenceNumber, model.SelectedStandard.Title, versions, application.ApplicationType);
 
             return RedirectToAction("SequenceSignPost", "Application", new { Id = id });
         }
 
         [HttpGet("standard/{id}/opt-in/{standardReference}/{version}")]
         [ApplicationAuthorize(routeId: "Id")]
-        public async Task<IActionResult> OptIn(Guid id, string standardReference, string version)
+        public async Task<IActionResult> OptIn(Guid id, string standardReference, decimal version)
+        {
+            var application = await _applicationApiClient.GetApplication(id);
+           
+            if (!CanUpdateApplicationAsync(application))
+            {
+                return RedirectToAction("Applications", "Application");
+            }
+
+            var standards = await _standardVersionApiClient.GetStandardVersionsByIFateReferenceNumber(standardReference);
+            var stdVersion = standards.First(x => x.Version.Equals(version.ToString(), StringComparison.InvariantCultureIgnoreCase));
+
+            var model = new StandardOptInViewModel()
+            {
+                Id = id,
+                StandardReference = stdVersion.IFateReferenceNumber,
+                StandardTitle = stdVersion.Title,
+                Version = stdVersion.Version,
+                EffectiveFrom = stdVersion.VersionEarliestStartDate ?? DateTime.Today,
+                EffectiveTo = stdVersion.VersionLatestEndDate
+            };
+
+            return View("~/Views/Application/Standard/OptIn.cshtml", model);
+        }
+
+        [HttpPost("standard/{id}/opt-in/{standardReference}/{version}")]
+        public async Task<IActionResult> OptInPost(Guid id, string standardReference, decimal version)
         {
-            var application = await _apiClient.GetApplication(id);
+            var application = await _applicationApiClient.GetApplication(id);
+            var contact = await GetUserContact();
 
             if (!CanUpdateApplicationAsync(application))
             {
                 return RedirectToAction("Applications", "Application");
             }
 
-            return View("~/Views/Application/Standard/OptIn.cshtml");
+            var org = await _orgApiClient.GetEpaOrganisation(application.OrganisationId.ToString());
+            var standards = await _standardVersionApiClient.GetStandardVersionsByIFateReferenceNumber(standardReference);
+            var stdVersion = standards.First(x => x.Version.Equals(version.ToString(), StringComparison.InvariantCultureIgnoreCase));
+
+            await _orgApiClient.OrganisationStandardVersionOptIn(id, contact.Id, org.OrganisationId, standardReference, version, stdVersion.StandardUId, null);
```

Comment:
Should comments be set here to something like "Opted in by EPAO" so we know how it occurred in the DB rather than null?

Reply:
The DateVersionApproved is null in the Db, it probably shouldn't be set on the request, but inside the opt in handler, should that be set to the current date the opt in occurred as when the record appears, they are "approved" for the version?

Reply:
> 
> 
> Should comments be set here to something like "Opted in by EPAO" so we know how it occurred in the DB rather than null?

MAybe could include the username "Opted in by xxxxxxxx"

Reply:
Ive updated the comments to "Opted in by EPAO" as suggested. I purposely left the DateVersionApproved as NULL as the opt-in does not have an approval process.